README.md

![flutter_cxui](assets/static/cxshow.png)

# flutter_cxui

<!-- flutter_cxui是一个基于Flutter框架的UI工具包，开发在Material框架下。它提供了丰富的可重用组件和样式， 它可以帮助开发者快速构建高性能、高保真度、高一致性的跨平台应用程序。 -->
flutter_cxui is a flutter ui kit with material theme. It includes a lot of reusable components and styles, it can help developers quickly build high-performance, high-correctness, and high-consistency applications.

## Install

```shell
  flutter pub add flutter_cxui
```

or

```yaml
dependencies:
  flutter_cxui: [version]
```

**Jump to [`Flutter_cxui`](https://pub.dev/packages/flutter_cxui) package.**

## Usage

`CxConfig` is custom class.

```Dart
import 'package:flutter_cxui/cxui.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body:  CxCard(
          shadow: true,
          radius: 16,
          bgColor: CxConfig.primary,
          hdBgColor: CxConfig.white.withAlpha(90),
          titleColor: CxConfig.white,
          margin: EdgeInsets.all(10),
          title: "Actions",
          actions: [
            CxIconButton(
              icon: Icons.arrow_circle_right_outlined,
              color: CxConfig.white,
              size: 18,
            )
          ],
          body: Text("hello"),
        ),
      ),
    );
  }
}
```

## Components

<!-- ![选择按钮列表](assets/static//select_button_list.jpg) -->

|||
|--|--|
|![home](assets/static/list.png)|![place_grid](assets/static/place_grid.png)|
|![icon_button](assets/static/icon_button.png)|![select_button_list](assets/static/select_button_list.png)|
|![image_card](assets/static/image_card.png)|![card](assets/static/card.png)|
|![title_nav](assets/static/title_nav.png)|![slider](assets/static/slider.jpg)|

## Demo

|        |   |
| ----------- | ----------- |
| ![Movie Home](assets/static/movie_home.png)     | ![Movie Item](assets/static/movie_item.png) |

## Thanks for supporting us

<!-- ![微信支持](assets/static/wechat.jpg) -->
![赞赏支持](assets/static/reward.jpg)


CHANGELOG.md
# FLUTTER CXUI

## 1.0.1+1

  Improve documentation comments for main classes.

## 1.0.1

1. 修改项目结构
2. 修复CxCard的const问题

## 1.0.0+1

1. ICON BUTTON的完成
2. TITLE NAV的完成
3. SELECT BUTTON LIST的完成
4. Card的基本完成
5. IMAGE CARD的完成
6. Slider View的完成
7. Place Grid的完成


example/lib/main.dart
import 'package:flutter/material.dart';
import './config.dart';
import './pages/app/movie/page_movie_home.dart';
import './pages/app/movie/page_movie_item.dart';
import './pages/page_card.dart';
import './pages/page_icon_button.dart';
import './pages/page_place_grid.dart';
import './pages/page_select_button_list.dart';
import './pages/page_slider_view.dart';
import './pages/page_title_nav.dart';
import 'package:ionicons/ionicons.dart';

import 'pages/page_image_card.dart';

class RouteItem {
  String path;
  String name;
  RouteItem(this.path, this.name);
}

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(
          seedColor: Colors.purple,
          background: CxConfig.background,
          primary: CxConfig.primary,
        ),
        appBarTheme: AppBarTheme(
          titleTextStyle: const TextStyle(fontSize: 18),
          backgroundColor: CxConfig.primary,
          shadowColor: Colors.transparent,
          // shape: const RoundedRectangleBorder(
          //   borderRadius: BorderRadius.only(
          //     bottomLeft: Radius.circular(20),
          //     bottomRight: Radius.circular(20),
          //   ),
          // ),
        ),
      ),
      // onGenerateRoute: (RouteSettings setting) {

      // },
      routes: {
        "/place_grid": (context) => const PagePlaceGrid(),
        "/icon_button": (context) => const PageIconButton(),
        "/select_button_list": (context) => const PageSelectButtonList(),
        "/image_card": (context) => const PageImageCard(),
        "/title_nav": (context) => const PageTitleNav(),
        "/card": (context) => const PageCard(),
        "/slider_view": (context) => const PageSliderView(),
        "/app/movie/home": (context) => const PageMovieHome(),
        "/app/movie/item": (context) => const PageMovieItem(),
      },
      title: 'CXUI',

      home: const MyHomePage(title: ''),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  List<RouteItem> routes = [
    RouteItem("/place_grid", "Place Grid"),
    RouteItem("/icon_button", "Icon Button"),
    RouteItem("/select_button_list", "Select Button List"),
    RouteItem("/title_nav", "Title Nav"),
    RouteItem("/image_card", "ImageCard"),
    RouteItem("/card", "Card"),
    RouteItem("/slider_view", "Slider View"),
    RouteItem("/app/movie/home", "Movie"),
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        // shadowColor: Colors.transparent,
        backgroundColor: CxConfig.primary,
        clipBehavior: Clip.antiAlias,
        leading: const Icon(Ionicons.apps_outline, size: 20),
        centerTitle: true,
        title: Image.asset(
          "assets/img/cx-logo.png",
          width: 60,
        ),
        actions: const [
          Icon(
            Ionicons.notifications_outline,
            size: 20,
          ),
          SizedBox(
            width: 10,
          ),
        ],
      ),
      body: SingleChildScrollView(
        child: Column(
          children: [
            Container(
              // height: 100,
              width: double.infinity,
              padding: const EdgeInsets.only(
                  top: 0, bottom: 15, right: 15, left: 15),
              decoration: BoxDecoration(
                color: CxConfig.primary,
                borderRadius: const BorderRadius.only(
                  bottomLeft: Radius.circular(20),
                  bottomRight: Radius.circular(20),
                ),
              ),
              child: Row(
                children: [
                  Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      Text(
                        "Hi CXUI",
                        style: TextStyle(
                          fontSize: 18,
                          color: Colors.white.withAlpha(230),
                        ),
                      ),
                      const SizedBox(
                        height: 5,
                      ),
                      Text(
                        "A New Flutter UI",
                        style: TextStyle(
                          color: CxConfig.white,
                          fontWeight: FontWeight.bold,
                          fontSize: 22,
                        ),
                      ),
                    ],
                  ),
                ],
              ),
            ),
            Container(
              margin: const EdgeInsets.only(top: 20),
              child: ListView.separated(
                physics: const NeverScrollableScrollPhysics(),
                padding: const EdgeInsets.symmetric(horizontal: 10),
                shrinkWrap: true,
                itemBuilder: itemBuilder,
                separatorBuilder: separatorBuilder,
                itemCount: routes.length,
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget? itemBuilder(BuildContext context, int index) {
    return Container(
      clipBehavior: Clip.antiAlias,
      // margin: EdgeInsets.symmetric(vertical: 8, horizontal: 10),
      decoration: const BoxDecoration(
        color: Colors.white,
        // boxShadow: [
        //   BoxShadow(
        //     color: CxConfig.primary.withOpacity(0.1),
        //     spreadRadius: 5,
        //     blurRadius: 7,
        //     offset: Offset(4, 8), // changes position of shadow
        //   ),
        // ],
        borderRadius: BorderRadius.all(Radius.circular(10)),
      ),
      child: ListTile(
        // tileColor: Colors.white,
        // selectedTileColor: Colors.amber,

        hoverColor: Colors.transparent,
        // style: ListTileStyle.list,
        // shape: const RoundedRectangleBorder(
        // borderRadius: BorderRadius.all(Radius.circular(10))),
        leading: Text(routes[index].name),
        trailing: Container(
          padding: const EdgeInsets.all(5),
          decoration: BoxDecoration(
            color: const Color(0xFFFF8A01),
            borderRadius: BorderRadius.circular(20),
          ),
          child: const Icon(
            Icons.arrow_forward_ios,
            size: 12,
            color: Colors.white,
          ),
        ),
        //hoverColor: Colors.green,
        //focusColor: Colors.green[100],
        onTap: () {
          Navigator.pushNamed(context, routes[index].path);
        },
      ),
    );
  }

  Widget separatorBuilder(BuildContext context, int index) {
    // return Container(height: 0, color: Colors.grey[200]);
    return Container(height: 16, color: Colors.transparent);
  }
}
